specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,176 @@
+for caption controls and resize regions. 
+
+For security and flexibility, we want developers to be able to enable or disable all 
+custom non-client functionality per WebView. Non-client functionality will affect that 
+app window’s size and position so it’s important that developers can definitively 
+toggle access. This can be achieved in a limited way using a feature flag, but feature 
+flags are applied per WebView2 environment, thus, an API on the WebView2 to enable/disable 
+non-client support via a setting is the better solution.
+
+# Description
+`IsNonClientRegionSupportEnabled` defaults to `FALSE`. Disabling/Enabling 
+`IsNonClientRegionSupportEnabled` takes effect after the next navigation. Currently, draggable 
+regions is the only non-client experience we have implemented. Eventually, this setting will 
+expand to enable other non-client functionality, such as resize and caption controls. 
+
+When the setting is set to `TRUE`, then the following non-client region support for only the top 
+level document will be enabled:  
+* Web pages will be able to use the `app-region` CSS style. 
+* Draggable Regions declared with the CSS style `app-region: drag` will support title bar 
+functionality, for example, the dragging of the app window, the title bar context menu upon 
+right click, and the maximizing/restoring of the window size upon double click of the html 
+element. 
+
+When set to `FALSE`, then all non-client region support will be disabled.  
+* Web pages will not be able to use the `app-region` CSS style. 
+
+# Examples
+This example enables non-client region support for all pages on www.microsoft.com. 
+Pages on other origins will not have non-client region support enabled. 
+
+## Win32 C++
+```cpp 
+ScenarioNonClientRegionSupport::ScenarioNonClientRegionSupport(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    
+    CHECK_FAILURE(m_webView->add_NavigationStarting(
+        Callback<ICoreWebView2NavigationStartingEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2NavigationStartingEventArgs* args) -> HRESULT
+            {
+                static const PCWSTR allowedHostName = L"www.microsoft.com";
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Uri(&uri));
+                wil::unique_bstr domain = GetDomainOfUri(uri.get());
+                
+                wil::com_ptr<ICoreWebView2Settings> m_settings;
+                CHECK_FAILURE(m_webView->get_Settings(&m_settings));
+                wil::com_ptr<ICoreWebView2Settings12> coreWebView2Settings12;
+                coreWebView2Settings12 = m_settings.try_query<ICoreWebView2Settings12();
+                CHECK_FEATURE_RETURN(coreWebView2Settings12);
+                
+                BOOL enabled;
+                CHECK_FAILURE(coreWebView2Settings12->get_IsNonClientRegionSupportEnabled(&enabled));
```

Comment:
Is there really a need to check the old value before setting the new one?

Could this be simplified to
```cpp
bool trusted = _wcsicmp(domain.get(), allowedHostName) == 0;
CHECK_FAILURE(coreWebView2Settings12->put_IsNonClientRegionSupportEnabled(trusted));
```

Reply:
Looks good. Please consider simplifying the sample code.

Reply:
changed